test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,241 @@
+  // This downcast will not overflow because size is capped by port numbers
+  uint32_t host_set_size = host_set.hosts_.size();
+  host_set.healthy_hosts_.clear();
+  host_set.degraded_hosts_.clear();
+  host_set.excluded_hosts_.clear();
+
+  Fuzz::ProperSubsetSelector subset_selector(random_bytestring);
+
+  std::vector<std::vector<uint8_t>> subsets = subset_selector.constructSubsets(
+      {num_healthy_hosts, num_degraded_hosts, num_excluded_hosts}, host_set_size);
+
+  // Healthy hosts are first subset
+  for (uint8_t index : subsets.at(0)) {
+    host_set.healthy_hosts_.push_back(host_set.hosts_[index]);
+    ENVOY_LOG_MISC(trace, "Index of host made healthy at priority level {}: {}",
+                   priority_of_host_set, index);
+  }
+
+  // Degraded hosts are second subset
+  for (uint8_t index : subsets.at(1)) {
+    host_set.degraded_hosts_.push_back(host_set.hosts_[index]);
+    ENVOY_LOG_MISC(trace, "Index of host made degraded at priority level {}: {}",
+                   priority_of_host_set, index);
+  }
+
+  // Excluded hosts are third subset
+  for (uint8_t index : subsets.at(2)) {
+    host_set.excluded_hosts_.push_back(host_set.hosts_[index]);
+    ENVOY_LOG_MISC(trace, "Index of host made excluded at priority level {}: {}",
+                   priority_of_host_set, index);
+  }
+
+  // Handle updating health flags for hosts_per_locality_
+  HostVector healthy_hosts_locality_one = {};
+  HostVector degraded_hosts_locality_one = {};
+  HostVector excluded_hosts_locality_one = {};
+
+  HostVector healthy_hosts_locality_two = {};
+  HostVector degraded_hosts_locality_two = {};
+  HostVector excluded_hosts_locality_two = {};
+
+  HostVector healthy_hosts_locality_three = {};
+  HostVector degraded_hosts_locality_three = {};
+  HostVector excluded_hosts_locality_three = {};
+
+  for (uint8_t index : subsets.at(0)) {
+    // If the host is in a locality, we have to add it to healthy hosts per locality
+    if (!(locality_indexes_.find(index) == locality_indexes_.end())) {
+      ENVOY_LOG_MISC(trace, "Added healthy host at index {} in locality {}", index,
+                     locality_indexes_[index]);
+      switch (locality_indexes_[index]) {
+      case 1:
+        healthy_hosts_locality_one.push_back(host_set.hosts_[index]);
+        break;
+      case 2:
+        healthy_hosts_locality_two.push_back(host_set.hosts_[index]);
+        break;
+      case 3:
+        healthy_hosts_locality_three.push_back(host_set.hosts_[index]);
+        break;
+      default: // shouldn't hit
+        NOT_REACHED_GCOVR_EXCL_LINE;
+      }
+    }
+  }
+
+  for (uint8_t index : subsets.at(1)) {
+    // If the host is in a locality, we have to add it to degraded hosts per locality
+    if (!(locality_indexes_.find(index) == locality_indexes_.end())) {
+      ENVOY_LOG_MISC(trace, "Added degraded host at index {} in locality {}", index,
+                     locality_indexes_[index]);
+      switch (locality_indexes_[index]) {
+      case 1:
```

Comment:
Prefer not hardcoding numbers, so maybe make this an integer `enum`.

Reply:
Changed to enum.